frontend/model/contracts/group.js

```diff
@@ -866,6 +867,8 @@ sbp('chelonia/defineContract', {
           // someone updated their income details, create a snapshot of the haveNeeds
           updateCurrentDistribution({ meta, state, getters })
         }
+
+        sbp('okTurtles.events/emit', GROUP_PROFILE_UPDATE)
```

Comment:
I just thought of another reason why it may more more sense to do it in the IncomeDetails modal when "OK" is clicked. Say you opened your income details and decided that in fact everything is correct as it looks, and so you click "OK" to confirm them. Then you're still "updating your income details" in the sense that you're confirming they're still correct, and so the reminder should disappear. We can't do that with this method, but doing in the IncomeDetails modal allows for it and is also simpler.

Reply:
@taoeffect Yeah, reading through these comments, It sounds like creating an event for a modal button click is a cleaner and sufficient enough approach. I will make a new change that way. 
Btw, This long detailed documentation on how a GI contract works in the app is very helpful in understanding which means what in the contracts. will take some to play around with them too. thanks.